parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -623,6 +627,30 @@ public ParquetFileReader(Configuration conf, Path file, ParquetMetadata footer)
     this.allocator = new HeapByteBufferAllocator();
   }
 
+  /**
+   * @param conf the Hadoop Configuration
+   * @param fileStatus FileStatus of a parquet file
+   * @param footer a {@link ParquetMetadata} footer already read from the file
+   * @throws IOException if the file can not be opened
+   */
+  public ParquetFileReader(Configuration conf, FileStatus fileStatus, ParquetMetadata footer) throws IOException {
+    this.converter = new ParquetMetadataConverter(conf);
+    this.conf = conf;
+    FileSystem fs = fileStatus.getPath().getFileSystem(conf);
+    this.fileStatus = fileStatus;
+    this.f = HadoopStreams.wrap(fs.open(fileStatus.getPath()));
```

Comment:
If Hadoop had an API to efficiently open a file plus return its file status and then the S3 and HDFS backends both used that, then I think we could avoid the subsequent open with the right passing-around of references.  But until that does I think we can't avoid those two calls